### PR TITLE
Add steps to lint GHA to start/stop docker container and run tests.

### LINKED ISSUE #14: Enable server side tests in Github Actions
**Description**
GitHub Actions should run server side tests as part of CI.  There's currently some [commented-out code](https://github.com/timmyichen/dev-directory/blob/main/.github/workflows/lint.yml#L39-L42) in the workflow that we can restore.  Before doing so, however, we'd need to serve a test database through docker on GitHub Actions.  Apparently it's just [supported](https://github.com/peter-evans/docker-compose-actions-workflow) on GHA so no additional config needed.

**Acceptance Criteria**
* Server tests run in CI

### PR DESCRIPTION
**Pull request by MatthewBozin**
Closes #14 

# Description

Add steps to lint GHA to start/stop docker container and run tests.

## Testing

If it passes CI it works

## Type of change

Please remove all except for everything applicable, and then remove this line.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] Changes have new/updated automated tests
- [ ] Changes have been manually tested
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by timmyichen — APPROVED** [review]
🎉

File: .env.test [hunk 1]
```diff
@@ -5,3 +5,7 @@ DB_PORT=5432
 DB_USER=postgres
 DB_HOST=localhost
 DB_NAME=test
+DISCORD_CLIENT_ID=123
+NEXT_PUBLIC_DISCORD_CLIENT_ID=123
+DISCORD_CLIENT_SECRET=123
+DB_PASSWORD=password
```
Comment (devonzara) [inline .env.test:11]: I'd try to keep these organized so they're easier to maintain. For instance, we should probably move `DB_PASSWORD` up to be with the other `DB_` vars. If we wanted to take it a step further, we could separate them into groups too. eg.

```ini
AUTH_SECRET=secret

DB_HOST=localhost
DB_PORT=5432
DB_NAME=test
DB_USER=postgres
DB_PASSWORD=password

DISCORD_CLIENT_ID=123
DISCORD_CLIENT_SECRET=123
DISCORD_REDIRECT_URI=http://localhost:3000/auth-redirect
DISCORD_GUILD_ID=735923219315425401

NEXT_PUBLIC_DISCORD_CLIENT_ID=123
```

**Review by devonzara — CHANGES_REQUESTED** [review]

File: .env.test (hunk 1 above)
Comment (timmyichen) [reply .env.test:11]: Agree with this, but I would instead recommend grouping it by what's a secret / what's not a secret.

Comment (devonzara) [reply .env.test:11]: @timmyichen, do you meaning something like this?
```ini
# Secrets
AUTH_SECRET=secret
DISCORD_CLIENT_SECRET=123
DB_PASSWORD=password

# Configuration Values
DB_HOST=localhost
DB_PORT=5432
DB_NAME=test
DB_USER=postgres

DISCORD_CLIENT_ID=123
DISCORD_REDIRECT_URI=http://localhost:3000/auth-redirect
DISCORD_GUILD_ID=735923219315425401

NEXT_PUBLIC_DISCORD_CLIENT_ID=123
```

Comment (timmyichen) [reply .env.test:11]: yep!

Comment (MatthewBozin) [reply .env.test:11]: Sounds good, will make the change!

**Review by timmyichen — APPROVED** [review]

**Review by wo1vin — APPROVED** [review]

**Review by devonzara — APPROVED** [review]

**Comment by devonzara** [comment]
@MatthewBozin, overlooked this before, but was there a specific need for the additional `docker-compose.test.yml`?

**Pull request merged**